sources/cassandra/validations/validation.go

```diff
@@ -0,0 +1,498 @@
+						mu.Lock()
+						errors++
+						mu.Unlock()
+					}
+					continue
+				}
+
+				var mismatchDetails MismatchDetail
+				diffFound, mismatchDetails := diffRows(sourceRow, targetRow)
+				if !diffFound {
+					continue
+				}
+				if mismatchDetails.MissingInTarget {
+					mu.Lock()
+					missing++
+					fmt.Printf("MISSING: row in target for source row: %+v\n", sourceRow)
+					mu.Unlock()
+				} else {
+					mu.Lock()
+					mismatches++
+					fmt.Printf("MISMATCH: found for row: %+v\n", mismatchDetails)
+					mu.Unlock()
+
+				}
+			}
+		}()
+	}
+	wg.Wait()
+	return errors, mismatches, missing
+}
+
+// diffRows compares a row from the source cluster with its corresponding row in the target cluster.
+// It checks for both missing rows and value mismatches in all columns.
+// Returns:
+// - bool: true if any difference is found
+// - MismatchDetail: details of the mismatch if found
+func diffRows(sourceRow, targetRow map[string]interface{}) (bool, MismatchDetail) {
+	// If target row is nil, it's missing in target
+	if targetRow == nil || len(targetRow) == 0 {
+		return true, MismatchDetail{
+			SourceRow:       sourceRow,
+			TargetRow:       nil,
+			MissingInTarget: true,
+		}
+	}
+
+	// Compare all column values
+	mismatch := false
+	for col, sourceVal := range sourceRow {
+		targetVal, exists := targetRow[col]
+		if !exists || !compareValues(sourceVal, targetVal) {
+			mismatch = true
+			break
+		}
+	}
+
+	if mismatch {
+		return true, MismatchDetail{
+			SourceRow:       sourceRow,
+			TargetRow:       targetRow,
+			MissingInTarget: false,
+		}
+	}
+
+	return false, MismatchDetail{}
+}
+
+// compareValues performs deep comparison of two values that may be of different types.
+// It handles special cases for various data types including:
+// - nil values
+// - byte arrays
+// - maps
+// - slices
+// For other types, it falls back to string representation comparison.
+// Returns true if the values are equal, false otherwise.
+func compareValues(v1, v2 interface{}) bool {
+	// Handle nil values
+	if v1 == nil && v2 == nil {
+		return true
+	}
+	if v1 == nil || v2 == nil {
+		return false
+	}
+
+	// Compare different types of values
+	switch val1 := v1.(type) {
+	case []byte:
+		if val2, ok := v2.([]byte); ok {
+			if len(val1) != len(val2) {
+				return false
+			}
+			for i := range val1 {
+				if val1[i] != val2[i] {
+					return false
+				}
+			}
+			return true
+		}
+	case map[string]interface{}:
+		if val2, ok := v2.(map[string]interface{}); ok {
+			if len(val1) != len(val2) {
+				return false
+			}
+			for k, v := range val1 {
+				if !compareValues(v, val2[k]) {
+					return false
+				}
+			}
+			return true
+		}
+	case []interface{}:
+		if val2, ok := v2.([]interface{}); ok {
+			if len(val1) != len(val2) {
+				return false
+			}
+			for i := range val1 {
+				if !compareValues(val1[i], val2[i]) {
+					return false
+				}
+			}
+			return true
+		}
+	}
```

Comment:
make sure you don't need a default case here.

Reply:
Yes primitive types have been verified.